src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -613,5 +667,34 @@ internal ImmutableArray<CompletionProvider> GetAllProviders(ImmutableHashSet<str
                     cancellationToken);
             }
         }
+
+        private class ProjectCompletionProvider
+            : AbstractProjectExtensionProvider<CompletionProvider, ExportCompletionProviderAttribute>
+        {
+            public ProjectCompletionProvider(AnalyzerReference reference)
+                : base(reference)
+            {
+            }
+
+            protected override bool SupportsLanguage(ExportCompletionProviderAttribute exportAttribute, string language)
+            {
+                return exportAttribute.Language == null
```

Comment:
Seems like all implementations for this method are the same. Consider extracting to a static helper in base type.

Reply:
The attributes are all different, so the methods look the same but the properties are all different.